resources/design.doc2.md

```diff
@@ -0,0 +1,181 @@
+
+U5. As a manager, I want to delete a task with a given ID.
+
+U6. As a manager, I want to view all projects I have created.
+
+U7. As a team member, I want to retrieve all tasks assigned to me.
+
+U8. As a team member, I want to retrieve all tasks assigned to me in a prioritized order (due date).
+
+U9. As a team member, I want to be able to update the progress of specific task (Not Started, In Progress, Complete).
+U10. As a team member, I want to be able to track the amount of time I spend on each task in order to accurately bill for my work or manage my own productivity.
+
+## 3. Stretch Goals
+U10. As a team member, I want to be able to track the amount of time I spend on each task in order to accurately bill for my work or manage my own productivity.
+
+U11. As a team member, I want to be able to comment on a specific task I am assigned to.
+
+U13. As a manager, I want to be able to share and collaborate files and documents with my team.
+
+## 4. Project Scope
+
+### 4.1. In Scope
+
+_Which parts of the problem defined in Sections 1 and 2 will you solve with this design? This should include the base functionality of your product. What pieces are required for your product to work?_
+
+_The functionality described above should be what your design is focused on. You do not need to include the design for any out of scope features or expansions._
+
+### 4.2. Out of Scope
+
+_Based on your problem description in Sections 1 and 2, are there any aspects you are not planning to solve? Do potential expansions or related problems occur to you that you want to explicitly say you are not worrying about now? Feel free to put anything here that you think your team can't accomplish in the unit, but would love to do with more time._
+
+_The functionality here does not need to be accounted for in your design._
+
+# 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+# 6. API
+
+## 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+
+```
+// ProjectModel
+String projectId;
+String projectName;
+String projectStatus;
+List<String> userIds;
+```
+
+```
+// TaskModel
+String taskId;
+String taskName;
+String taskDescription;
+String taskDueDate;
+String taskStatus;
+List<String> taskAssignedUsers;
+Integer taskTimer;
+```
+
+```
+// UserModel
+String userId;
+String userName;
+String userRole;
+List<String> taskIds;
+```
+
+## 6.2. _Create Project Endpoint_
```

Comment:
These should increment in number (e.g. this one is 6.2, the next should be 6.3, etc)